oshi-demo/src/main/java/oshi/demo/DetectVM.java

```diff
@@ -84,7 +84,7 @@ public static String identifyVM() {
         for (NetworkIF nif : nifs) {
             String mac = nif.getMacaddr().toUpperCase();
             String oui = findOuiByMacAddressIfPossible(mac);
-            if (oui != null) {
+            if (oui != null && oui != "") {
```

Comment:
You shouldn't use `!=` as a comparison for Strings, as they are objects.  The correct equality comparison would be `"".equals(oui)`.  However, it is slightly more efficient to do `!oui.isEmpty()`.